src/runtime_src/xdp/profile/database/static_info/xclbin_info.cpp

```diff
@@ -94,17 +145,568 @@ namespace xdp {
+      }
+      return nullptr ;
+    }
+
+    std::vector<Monitor*>* ConfigInfo::getAIMonitors(XclbinInfo* xclbin)
+    {
+      for (auto bin : currentXclbins) {
+        if (bin == xclbin)
+          return &(bin->pl.aims) ;
+      }
+      return nullptr ;
+    }
+
+    std::vector<Monitor*>* ConfigInfo::getASMonitors(XclbinInfo* xclbin)
+    {
+      for (auto bin : currentXclbins) {
+        if (bin == xclbin)
+          return &(bin->pl.asms) ;
+      }
+      return nullptr ;
+    }
+
+    std::vector<Monitor*> ConfigInfo::getUserAIMsWithTrace(XclbinInfo* xclbin)
+    {
+      std::vector<Monitor*> constructed ;
+      for (auto bin : currentXclbins) {
+        if (bin == xclbin) {
+          for (auto aim : bin->pl.aims) {
+            if (aim->traceEnabled && !aim->isShellMonitor())
+              constructed.push_back(aim) ;
+          }
+        }
+      }
+      return constructed ;
+    }
+
+    std::vector<Monitor*> ConfigInfo::getUserASMsWithTrace(XclbinInfo* xclbin)
+    {
+      std::vector<Monitor*> constructed ;
+      for (auto bin : currentXclbins) {
+        if (bin == xclbin) {
+          for (auto mon : bin->pl.asms) {
+            if (mon->traceEnabled && !mon->isShellMonitor())
+              constructed.push_back(mon) ;
+          }
+        }
+      }
+      return constructed ;
+    }
+
+    void ConfigInfo::addTraceGMIO(uint32_t id, uint8_t col, uint8_t num,
+                                uint8_t stream, uint8_t len)
+    {
+      for(auto xclbin : currentXclbins)
+      {
+        if(xclbin->aie.valid)
+        {
+          xrt_core::message::send(xrt_core::message::severity_level::info, "XRT", "added GMIO trace: "+ std::to_string(id));
+          xclbin->aie.gmioList.push_back(new TraceGMIO(id, col, num, stream, len)) ;
+          return ;
+        }
+      }
+    }
+
+    void ConfigInfo::addAIECounter(uint32_t i, uint8_t col, uint8_t r,
+                                   uint8_t num, uint16_t start, uint16_t end,
+                                   uint8_t reset, uint32_t load, double freq,
+                                   const std::string& mod,
+                                   const std::string& aieName)
+    {
+      for(auto xclbin : currentXclbins)
+      {
+        if(xclbin->aie.valid)
+        {
+          xclbin->aie.aieList.push_back(new AIECounter(i, col, r, num, start, end,
+                                                    reset, load, freq, mod, aieName)) ;
+          return ;
+        }
+      }
+    }
+
+    void ConfigInfo::addAIECounterResources(uint32_t numCounters,
+                                            uint32_t numTiles,
+                                            uint8_t moduleType)
+    {
+      XclbinInfo* xclbin = nullptr ;
+      for(auto bin : currentXclbins)
+      {
+        if(bin->aie.valid)
+        {
+          xclbin = bin;
+          break ;
+        }
+      }
+
+      if(!xclbin)
+        return;
+
+      switch (moduleType)
+      {
+        case 0:
```

Comment:
Is there an enumeration for this?

Reply:
addressed via https://github.com/Xilinx/XRT/pull/8244